### PR TITLE
feat: Adding the first version of the user creation in STP

### PR DESCRIPTION
**Pull request by Extor3D**
This PR adds all the logic to create a user in STP

- A REST endpoint in the service to create it
- The Webclient to connect into STP
- CLABE generation
- Business logic

A few things that still aren't implemented:

- We still dont have the certificates to sign the call
- The get method is still pending implementation, this also includes the update of the account registry when it already exists

image
Current error message when trying to create the user

image
Bad request error handling

### CODE REVIEW

File: src/main/kotlin/com/kamet/payment/core/adapter/persistence/ReadAccountAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@Mapper
+class ReadAccountAdapter(
+    val accountEntityMapper: AccountEntityMapper,
+    val accountRepository: AccountRepository,
+) : ReadAccountPort {
+    val sequences =
+        mutableMapOf<Int, Int>(
+            Pair(Client.KAMET.id, 0),
+            Pair(Client.OTHER.id, 0),
+        )
```
Comment (juan1756) [inline src/main/kotlin/com/kamet/payment/core/adapter/persistence/ReadAccountAdapter.kt:19]:
```suggestion
        mutableMapOf {
         Client.KAMET.id to 0,
         Client.OTHER.id to 0
        }
```

File: src/main/kotlin/com/kamet/payment/core/domain/model/Client.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.kamet.payment.core.domain.model
+
+enum class Client(val id: Int) {
```
Comment (juan1756) [inline src/main/kotlin/com/kamet/payment/core/domain/model/Client.kt:3]: Buenisimo hardcodear esto, mas adelante cuando haga falta migrar Cashmind lo hacemos modelo

File: src/main/kotlin/com/kamet/payment/core/domain/stp/StpStatus.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.kamet.payment.core.domain.stp
+
+enum class StpStatus(val id: Number) {
+    REVISION(0),
+    INVALID_SIGN(1),
+    EXCEEDS_BATCH(2),
+    DUPLICATED_ACCOUNT(3),
+    COUNTRY_REQUIRED(4),
+    MOTHER_SURNAME_REQUIRED(5),
```
Comment (juan1756) [inline src/main/kotlin/com/kamet/payment/core/domain/stp/StpStatus.kt:9]: Is this a STP state? weird, sounds more like a error reason

**Review by juan1756 — APPROVED** [review]

File: src/main/kotlin/com/kamet/payment/core/domain/stp/StpStatus.kt (hunk 3 above)
Comment (Extor3D) [reply src/main/kotlin/com/kamet/payment/core/domain/stp/StpStatus.kt:9]: I should rename it to something more explicit. It's the response code you can get when creating a User in STP. I'll rename it ASAP

Comment (Extor3D) [reply src/main/kotlin/com/kamet/payment/core/domain/stp/StpStatus.kt]: Renamed the class to be more explicit

**Review by lucianoclusa — APPROVED** [review]

**Review by juan1756 — APPROVED** [review]

**Review by iLeonelPerea — APPROVED** [review]

**Pull request merged**